### PR TITLE
Add second row of options to dm voting

### PR DESCRIPTION
**Pull request by HKGx**
This pull request introduces significant changes to the `dmVoting` feature, including the addition of row-based options for polls, improvements to the user interface, and the implementation of a utility function for converting numbers to emojis. The most important changes are summarized below:

### Enhancements to Poll Options:

* [`apps/bot/src/dmVoting.ts`](diffhunk://#diff-e1a6ba594b862da64daadd05316763833ab8adf08872225ed0c8d2f9fb7c2660L50-R82): Modified the poll creation and update process to support two rows of options, with validation to ensure no more than five options per row. [[1]](diffhunk://#diff-e1a6ba594b862da64daadd05316763833ab8adf08872225ed0c8d2f9fb7c2660L50-R82) [[2]](diffhunk://#diff-e1a6ba594b862da64daadd05316763833ab8adf08872225ed0c8d2f9fb7c2660L113-R178) [[3]](diffhunk://#diff-e1a6ba594b862da64daadd05316763833ab8adf08872225ed0c8d2f9fb7c2660L186-R274)
* [`apps/bot/src/dmVoting.ts`](diffhunk://#diff-e1a6ba594b862da64daadd05316763833ab8adf08872225ed0c8d2f9fb7c2660R315-R331): Updated the display and formatting of poll options to differentiate between rows and added appropriate labels. [[1]](diffhunk://#diff-e1a6ba594b862da64daadd05316763833ab8adf08872225ed0c8d2f9fb7c2660R315-R331) [[2]](diffhunk://#diff-e1a6ba594b862da64daadd05316763833ab8adf08872225ed0c8d2f9fb7c2660L297-R376)

### Utility Function:

* [`apps/bot/src/util/numberToEmoji.ts`](diffhunk://#diff-1d62c9feaedea4485b3bcf43212bd546ae18674bc8d527e123a9cace4c99e12aR1-R19): Added a new utility function `numberToEmoji` to convert numbers to their corresponding emoji representations.
* [`apps/bot/test/numberToEmoji.test.ts`](diffhunk://#diff-35f8a97baf03389410575aeb0eefccda342b9952843691d0af806d34ae701aedR1-R17): Created unit tests for the `numberToEmoji` function to ensure correct functionality.

### Database Schema Changes:

* [`packages/db/prisma/migrations/20241114181515_add_row_to_dm_poll_option/migration.sql`](diffhunk://#diff-c7fab10a4ec8b6ffd7ffb1e99a3ad93641cbe806d1524d4b885ced5bb445221bR1-R2): Added a new `row` column to the `DmPollOption` table to support row-based poll options.
* [`packages/db/prisma/migrations/20241114181534_remove_default_from_dm_poll_option_row/migration.sql`](diffhunk://#diff-456e64ea9c57d38f6245918bf8bcc62102a1f69876c184c054ec0f843320a823R1-R2): Removed the default value from the `row` column in the `DmPollOption` table.
* [`packages/db/prisma/schema/dmVoting.prisma`](diffhunk://#diff-8a8ee75562ebc88efd676500986714e5435cc5a8aeb6c098869b782de689e2beR19): Updated the `DmPollOption` model to include the new `row` field.

### Code Improvements:

* [`apps/bot/src/dmVoting.ts`](diffhunk://#diff-e1a6ba594b862da64daadd05316763833ab8adf08872225ed0c8d2f9fb7c2660L395-R496): Replaced the hardcoded number emojis with the `numberToEmoji` function to dynamically generate emojis for poll options.
* [`apps/bot/src/dmVoting.ts`](diffhunk://#diff-43abb4d0e6ed03f809cd62ea6325506cae01979ca41e9591556372f8c3f12b79R100): Added a TODO comment to decouple logger registration from guild creation.

### CODE REVIEW

File: apps/bot/src/dmVoting.ts [hunk 1]
```diff
@@ -47,22 +48,38 @@ const getPollCreateOrUpdateActionRows = (poll: DMPollWithOptions | null = null)
...
     .setMaxLength(256)
     .setStyle(TextInputStyle.Paragraph);
 
+  const secondRowInput = new TextInputBuilder()
+    .setCustomId("row2")
+    .setLabel("Drugi rząd")
+    .setPlaceholder("Pusty głos\nWycofaj z przyszłych głosowań")
+    .setRequired(false)
+    .setMinLength(2)
+    .setMaxLength(256)
```
Comment (Daste745) [inline apps/bot/src/dmVoting.ts:66]: Powinniśmy pozwolić zostawić drugi wiersz jako pusty dla głosowań gdzie nie jest on potrzebny.

File: apps/bot/src/dmVoting.ts [hunk 2]
```diff
@@ -183,32 +223,55 @@ export const dmVoting = new Hashira({ name: "dmVoting" })
             const content = submitAction.components
               .at(1)
               ?.components.find((c) => c.customId === "content")?.value;
-            const rawOptions = submitAction.components
+            const rawFirstRowOptions = submitAction.components
               .at(2)
-              ?.components.find((c) => c.customId === "options")?.value;
-            if (!title || !content || !rawOptions) {
+              ?.components.find((c) => c.customId === "raw1")?.value;
```
Comment (Daste745) [inline apps/bot/src/dmVoting.ts:228]: Typo
```suggestion
              ?.components.find((c) => c.customId === "row1")?.value;
```

File: apps/bot/src/dmVoting.ts [hunk 3]
```diff
@@ -183,32 +223,55 @@ export const dmVoting = new Hashira({ name: "dmVoting" })
...
+            const rawSecondRowOptions = submitAction.components
+              .at(3)
+              ?.components.find((c) => c.customId === "raw2")?.value;
```
Comment (Daste745) [inline apps/bot/src/dmVoting.ts:231]: Typo
```suggestion
              ?.components.find((c) => c.customId === "row2")?.value;
```

File: apps/bot/src/dmVoting.ts (hunk 1 above)
Comment (HKGx) [reply apps/bot/src/dmVoting.ts:66]: Jest `setRequired(false)`

**Pull request merged**